neqo-transport/src/connection/tests/ecn.rs

```diff
@@ -0,0 +1,77 @@
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![warn(clippy::pedantic)]
+
+use neqo_common::IpTos;
+use test_fixture::now;
+
+use crate::{
+    cc::MAX_DATAGRAM_SIZE,
+    connection::tests::{connect_force_idle, default_client, default_server},
+    path::MAX_PATH_PROBES,
+    StreamType,
+};
+
+// This is more data than will fit into `MAX_PATH_PROBES` packets.
+const DATA_CLIENT: &[u8] = &[0; MAX_PATH_PROBES * MAX_DATAGRAM_SIZE];
+
+#[test]
+fn disables_when_bleached() {
+    let mut client = default_client();
+    let mut server = default_server();
+    connect_force_idle(&mut client, &mut server);
+    assert!(client.paths.primary().borrow().is_ecn_enabled());
+
+    let stream_id = client.stream_create(StreamType::BiDi).unwrap();
+    client.stream_send(stream_id, DATA_CLIENT).unwrap();
+
+    while let Some(mut client_pkt) = client.process_output(now()).dgram() {
+        // Bleach the ECN bits on the packet.
+        client_pkt.set_tos(IpTos::default());
+        server.process_input(&client_pkt, now());
+    }
+
+    // Client should now process ACKs with incorrect ECN counts and disable ECN.
+    while let Some(server_pkt) = server.process_output(now()).dgram() {
+        client.process_input(&server_pkt, now());
+    }
+
+    // ECN should now be disabled.
+    assert!(!client.paths.primary().borrow().is_ecn_enabled());
```

Comment:
Rather than poke around in the internals, just get the client to `send_something()` and check the TOS on those datagrams.